### PR TITLE
Support for product version supplied by AppVeyor

### PR DESCRIPTION
**Pull request by ge0rdi**
All binaries and installer itself will use product version supplied by _AppVeyor_ build.
Manual build (either in VS or with local build scripts) will use fixed version - 4.4.0.

Regarding _CS_VERSION_NUM_ related changes - this value is written to registry during installation and then used at one place (though seems only for some logging purposes). I had to change its format a bit so that it can encode _build_ number bigger than 255 (other places in code use similar format).

### CODE REVIEW

**Comment by smz** [comment]
Hello everybody and sorry for dropping in..

As far as regards version numbers, I suggest you to stick to the "Semantic Versioning".

If you're not familiar with the concept, have a look here: https://semver.org/

Good job keeping Classic Shell alive! Thanks!

**Comment by ge0rdi** [comment]
@smz 
_Semantic Versioning_ definitely makes sense.
Do you see an issue related to versioning in this PR?

I think we should change _minor_ to make it distinct from original _Classic Shell_ releases (4.3.x). Even if there weren't functional changes.

**Comment by smz** [comment]
TBH I haven't looked inside the PR: I'm just lurking around as I'm very interested in Classic Shell an having it surviving.

> I think we should change minor to make it distinct from original Classic Shell releases (4.3.x). Even if there weren't functional changes.

That's probably correct. But if you are going to "release" it, first please be sure to have the rights to the **name** "Classic Shell": forking an open source software doen't necessarily imply having the rights to using the same name (most times id toes not...). Check with the original developer if that's ok with him, and be sure nobody else is "trademarking" it...

**Comment by coddec** [comment]
@smz @ge0rdi 
Permission of naming issue: Checking with Ivo.
Issue https://github.com/coddec/Classic-Shell/issues/13

File: ClassicShellSrc/ClassicShellSetup/ClassicShellSetup.wxs [hunk 1]
```diff
@@ -7,20 +7,18 @@
...
 <?else ?>
-<?define CS_GUID = "$(env.CS_GUID32)"?>
 <?define CS_UPGRADE = "EB7DA1D9-44EA-43BF-8A08-0B103007D4F2"?>
 <?define CS_PLATFORM  = "x86"?>
 <?define CS_PROGRAM_FILES  = "ProgramFilesFolder"?>
 <?define CS_WIN64  = "no"?>
 <?endif ?>
 <Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
-    <Product Id="$(var.CS_GUID)" Name="Classic Shell" Language="1033" Version="$(env.CS_VERSION)" Manufacturer="IvoSoft" UpgradeCode="$(var.CS_UPGRADE)">
+    <Product Id="*" Name="Classic Shell" Language="1033" Version="$(env.CS_VERSION)" Manufacturer="IvoSoft" UpgradeCode="$(var.CS_UPGRADE)">
```
Comment (XenHat) [inline ClassicShellSrc/ClassicShellSetup/ClassicShellSetup.wxs]: I feel like we should do something about `Manufacturer` as IvoSoft ceased developing.
Depending on his reply to #13, we could change this to "The Classic Shell Open-Source Effort" or something like that?

Comment (ge0rdi) [reply ClassicShellSrc/ClassicShellSetup/ClassicShellSetup.wxs]: There are also _CompanyName_ and _LegalCopyright_ (`Copyright (C) 2009-2016, Ivo Beltchev`) fields in version info of binaries. And these probably have to be updated as well.
We can do that separately, based on Ivo's answer.

**Comment by ge0rdi** [comment]
@coddec 
I guess there is no need to wait for #13 with this PR. It is about ability to produce binaries/installer with version set in _AppVeyor_ build.
We can do other changes separately.

**Comment by coddec** [comment]
@ge0rdi I referenced #13 cause @smz mentioned the issue, but it's not necessary to wait #13 for this one, Since here's about "Support for product version supplied by AppVeyor" strictly speaking, not really relative. ;)

**Comment by ge0rdi** [comment]
@coddec 
If you are OK with changes in this PR, please, approve it so that I can merge it.

**Pull request merged**